ProcessLib/BoundaryCondition/NonuniformVariableDependantNeumannBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,131 @@
+            "CurrentVariableValues", _data.prefac_current_variable};
+        MeshNodeParameter<double> prefac_other_variable_values{
+            "OtherVariableValues", _data.prefac_other_variable};
+        MeshNodeParameter<double> prefac_mixed_variables_values{
+            "MixedVariablesValues", _data.prefac_mixed_variables};
+        // Get element nodes for the interpolation from nodes to
+        // integration point.
+        NodalVectorType const constant_node_values =
+            constant_values.getNodalValuesOnElement(Base::_element, t);
+        NodalVectorType const prefac_current_variable_node_values =
+            prefac_current_variable_values.getNodalValuesOnElement(
+                Base::_element, t);
+        NodalVectorType const prefac_other_variable_node_values =
+            prefac_other_variable_values.getNodalValuesOnElement(Base::_element,
+                                                                 t);
+        NodalVectorType const prefac_mixed_variables_node_values =
+            prefac_mixed_variables_values.getNodalValuesOnElement(
+                Base::_element, t);
+        unsigned const n_integration_points =
+            Base::_integration_method.getNumberOfPoints();
+        auto const indices_current_variable =
+            NumLib::getIndices(mesh_item_id, dof_table_boundary);
+        auto const indices_other_variable = NumLib::getIndices(
+            mesh_item_id, _data.dof_table_boundary_other_variable);
+        std::vector<double> const local_current_variable =
+            x.get(indices_current_variable);
+        std::vector<double> const local_other_variable =
+            x.get(indices_other_variable);
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            auto const& n_and_weight = Base::_ns_and_weights[ip];
+            auto const& N = n_and_weight.N;
+            auto const& w = n_and_weight.weight;
+
+            double current_variable_int_pt = 0.0;
+            double other_variable_int_pt = 0.0;
+
+            NumLib::shapeFunctionInterpolate(local_current_variable, N,
+                                             current_variable_int_pt);
+            NumLib::shapeFunctionInterpolate(local_other_variable, N,
+                                             other_variable_int_pt);
+            NodalVectorType const neumann_node_values =
+                constant_node_values +
+                prefac_current_variable_node_values * current_variable_int_pt +
+                prefac_other_variable_node_values * other_variable_int_pt +
+                prefac_mixed_variables_node_values * current_variable_int_pt *
+                    other_variable_int_pt;
+
+            _local_rhs.noalias() += N * neumann_node_values.dot(N) * w;
+        }
+
+        b.add(indices_current_variable, _local_rhs);
+    }
+
+private:
+    NonuniformVariableDependantNeumannBoundaryConditionData const& _data;
+    NodalVectorType _local_rhs;
```

Comment:
I'd think there is no need to store the rhs in the assembler. It is already passed in assemble(), just map it and don't set to zero.